tf-psa-crypto/tests/suites/test_suite_psa_crypto.function

```diff
@@ -10111,21 +10112,58 @@ void generate_key(int type_arg,
         TEST_ASSUME(status != PSA_ERROR_INSUFFICIENT_MEMORY);
     }
     TEST_EQUAL(status, expected_status);
-    if (expected_status != PSA_SUCCESS) {
-        goto exit;
+    if (expected_status == PSA_SUCCESS) {
+        /* Test the key information */
+        PSA_ASSERT(psa_get_key_attributes(key, &got_attributes));
+        TEST_EQUAL(psa_get_key_type(&got_attributes), type);
+        TEST_EQUAL(psa_get_key_bits(&got_attributes), bits);
+
+        /* Do something with the key according to its type and permitted usage. */
+        TEST_EQUAL(mbedtls_test_psa_exercise_key(key, usage, alg, 0), 1);
     }
 
-    /* Test the key information */
-    PSA_ASSERT(psa_get_key_attributes(key, &got_attributes));
-    TEST_EQUAL(psa_get_key_type(&got_attributes), type);
-    TEST_EQUAL(psa_get_key_bits(&got_attributes), bits);
+    /* Adjust expected_status for interruptible key generation.
+     * Interruptible key generation is only supported for ECC key pairs and even
+     * for those only when MBEDTLS_ECP_RESTARTABLE is on.
+     */
 
-    /* Do something with the key according to its type and permitted usage. */
-    if (!mbedtls_test_psa_exercise_key(key, usage, alg, 0)) {
-        goto exit;
+    if (!PSA_KEY_TYPE_IS_ECC(type)) {
+        expected_status = PSA_ERROR_NOT_SUPPORTED;
     }
 
+#if !defined(MBEDTLS_ECP_RESTARTABLE)
+    expected_status = PSA_ERROR_NOT_SUPPORTED;
+#endif
+
+    status = psa_generate_key_iop_setup(&operation, &attributes);
+    TEST_EQUAL(status, expected_status);
+
+    /* Test that calling setup 2 time consecutively fails */
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+    status = psa_generate_key_iop_setup(&operation, &attributes);
+    TEST_EQUAL(status, PSA_ERROR_BAD_STATE);
+#endif
+
+    PSA_ASSERT(psa_generate_key_iop_abort(&operation));
+
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+    /* In case the expected status is a failure test a valid input will succeed after abort */
+    if (expected_status != PSA_SUCCESS) {
+        memset(&attributes, 0, sizeof(attributes));
+        psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+        psa_set_key_bits(&attributes, 256);
```

Comment:
The CI is failing because we might not have this specific curve available. Every curve except CURVE448 and CURVE25519 are Weierstrass curves.

We could use a macro similar to [KNOWN_SUPPORTED_HASH_ALG](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/include/test/psa_exercise_key.h#L21-L45). Although we probably will need two macros, one for known supported family and one for known supported size.

Reply:
There are find-a-curve macros [in `test_suite_pk.function`](https://github.com/Mbed-TLS/mbedtls/blob/development/tf-psa-crypto/tests/suites/test_suite_pk.function#L64). Lifting them to a common file in `tests/include` is a bit of cleanup that we haven't gotten around to yet.

(I'm not sure if we need that here, as opposed to passing the curve as a test argument and having the corresponding dependency. Just saying that if we need the macros, they already exist, but not in the right file.)

Reply:
I think what we are trying to test here is test that setup works with a good input after failing before and calling abort() which I think was already tested before because in cases with valid parameters setup fails at first due to calling it two consecutive times and then we abort and call setup with the valid input so I don't think adding an equivalent of KNOWN_SUPPORTED_HASH_ALG is worth it in this case but I can still add it if you think otherwise.